.gitlab/docker/Dockerfile.spack_compiler

```
@@ -7,25 +7,22 @@ SHELL ["/bin/bash", "-c"]
 # CSCS CI's .container-builder
 RUN spack config --scope site add  concretizer:targets:host_compatible:false
 
+# Create an anonymous spack environment
+RUN spack env create pika_env
+
 # Install compiler if not already installed
 ARG ARCH
 ARG COMPILER
-RUN spack compiler info $COMPILER > /dev/null 2> /dev/null; compiler_missing=$?; \
+RUN spack -e pika_env compiler info $COMPILER > /dev/null 2> /dev/null; compiler_missing=$?; \
```

I think this can be
```
```suggestion
RUN spack compiler info $COMPILER > /dev/null 2> /dev/null; compiler_missing=$?; \
```
since the compiler can be added globally, not only in the environment.